executor/builder.go

```diff
@@ -2358,6 +2358,8 @@ func (b *executorBuilder) buildIndexLookUpJoin(v *plannercore.PhysicalIndexJoin)
 		if innerTypes[i].EvalType() == types.ETString {
 			innerTypes[i].Flen = types.UnspecifiedLength
 		}
+		// Use the probe table's collation.
+		outerTypes[i].Collate = innerTypes[i].Collate
```

Comment:
Do we need to deriveCollation here?

Reply:
No, we must use the probe table's collation.

Reply:
By the way, it causes Panic: index out of range.